src/api/PDFDocument.ts

```diff
@@ -692,6 +697,25 @@ export default class PDFDocument {
     return copiedPages;
   }
 
+  /**
+   * Attaches a file to a [[PDFDocument]] instance.
+   * @param file The input data containing the file to be attached.
+   * @param fileName Name of file to be attached.
+   */
+  attach(file: string | Uint8Array | ArrayBuffer, fileName: string): void {
```

Comment:
I'd like to add an options object to this method. There are a number of additional parameters that the spec allows us to define for attachments. These parameters are visible in the attachments table of Acrobat and users will find them to be very handy. For example:
```js
pdfDoc.attach(attachmentBytes, 'cool_stuff.pdf', {
  mimeType: 'application/pdf',
  description: 'This file contains some cool stuff! 😎',
  creationDate: new Date(),
  modificationDate: new Date(),
});
```
You can view the details for these parameters in section **7.11.3 - Table 44** and section **7.11.4 - Tables 45 and 46** of the PDF spec.

Note that you can use [`PDFHexString.fromText`](https://github.com/Hopding/pdf-lib/blob/master/src/core/objects/PDFHexString.ts#L17-L26) and [`PDFString.fromDate`](https://github.com/Hopding/pdf-lib/blob/master/src/core/objects/PDFString.ts#L20-L28) when converting the strings/dates to PDF objects in `PDFAttachmentEmbedder.ts`.

Reply:
Good idea! I implemented it with the following options, please let me know what you think. I'm not really sure about the size parameter, since the usual readers pick it up by default..
```js
{
  mimeType: 'application/pdf',
  description: 'This file contains some cool stuff! 😎',
  size: 2334,
  creationDate: new Date(),
  modificationDate: new Date(),
  checkSum: '61004f1f2cc449b9065d2e81764c9c00'
}
```